tiledb/sm/group/group.cc

```diff
@@ -72,16 +72,16 @@ Group::Group(const URI& group_uri, StorageManager* storage_manager)
     , changes_applied_(false) {
 }
 
-Status Group::open(
+void Group::open(
     QueryType query_type, uint64_t timestamp_start, uint64_t timestamp_end) {
   // Checks
   if (is_open_) {
-    return Status_GroupError("Cannot open group; Group already open");
+    throw Status_GroupError("Cannot open group; Group already open");
```

Comment:
We don't want to throw `Status` ever. It won't be handled correctly by the top-level API wrapper. This must be changed.

The correct class to throw is some class derived from `StatusException`. This can be done on a per-file basis (at the top of the file) or a per-directory basis (in a header not visible outside it). For a per-directory substitute, use `class GroupException : public StatusException`. If you want finer granularity, use `GroupException`, `GroupDetailsException`, `GroupDirectoryException`, etc.

Regardless, it appears that the function `Status_GroupError` can completely disappear by the end of this PR. See its implementation to observe that all it does is supply a constant origin string to a `Status` constructor. This pattern can be mimicked to implement `GroupException`.